### PR TITLE
Make kueue admin role bindings dynamic based on Auth CR

### PR DESCRIPTION
**Pull request by ugiordan**
## Description

This PR implements dynamic kueue admin role binding management within the kueue controller to replace hardcoded group configuration.

**Problem:**
The kueue admin role binding was managed by hardcoded groups (`rhods-admins`, `odh-admins`, `dedicated-admins`) in a static configuration file `config/kueue-configs/binding_admin_roles.yaml`. This approach violated architectural principles where each component should manage its own RBAC resources, and it didn't dynamically sync with the Auth CR admin groups configuration.

**Solution:**
- **Moved RBAC management from static config to kueue controller**: Created `manageKueueAdminRoleBinding()` action that dynamically reads admin groups from Auth CR
- **Added Auth CR watching**: Kueue controller now watches Auth CR changes to trigger reconciliation when admin groups are modified
- **Implemented defensive filtering**: Handles upgrade scenarios where Auth CRs might contain invalid groups (`system:authenticated`, empty strings) from before webhook validation was implemented
- **Removed static configuration**: Deleted `config/kueue-configs/binding_admin_roles.yaml` and updated kustomization
- **Added comprehensive testing**: Unit tests covering all scenarios including upgrade edge cases

**Key architectural improvements:**
- **Reactive Updates**: Kueue controller automatically responds to Auth CR changes
- **Better Separation**: Each component manages its own RBAC resources  
- **Dynamic Sync**: Kueue admin permissions stay synchronized with Auth CR configuration
- **Upgrade Safety**: Handles legacy Auth CRs that may contain invalid groups

**JIRA**: [RHOAIENG-26336](https://issues.redhat.com/browse/RHOAIENG-26336)

## How Has This Been Tested?

**Unit Tests Added:**
- `TestManageKueueAdminRoleBinding_AuthCRNotFound` - Tests graceful handling when Auth CR doesn't exist
- `TestManageKueueAdminRoleBinding_WithValidAdminGroups` - Tests normal case with valid admin groups
- `TestManageKueueAdminRoleBinding_WithFilteredAdminGroups` - Tests upgrade scenario with mixed valid/invalid groups
- `TestManageKueueAdminRoleBinding_WithOnlyInvalidAdminGroups` - Tests edge case where all groups are invalid
- `TestManageKueueAdminRoleBinding_WithEmptyAdminGroups` - Tests edge case with no admin groups

**Manual Testing:**
- Verified code compiles successfully: `go build ./internal/controller/components/kueue/...`
- All unit tests pass: `go test ./internal/controller/components/kueue/... -v`
- Tested filtering logic handles upgrade scenarios correctly

**Testing Coverage:**
- **Normal Operations**: Auth CR with valid admin groups creates correct ClusterRoleBinding
- **Upgrade Scenarios**: Legacy Auth CRs with invalid groups are properly filtered
- **Edge Cases**: Missing Auth CR, empty groups, only invalid groups all handled gracefully
- **Error Handling**: Auth CR fetch failures handled appropriately

## Screenshot or short clip

N/A - This is an internal RBAC management improvement without user-facing UI changes.

## Merge criteria

- [x] You have read the [contributors guide](https://github.com/opendatahub-io/opendatahub-operator/blob/incubation/CONTRIBUTING.md).
- [x] Commit messages are meaningful - have a clear and concise summary and detailed explanation of what was changed and why.
- [x] Pull Request contains a description of the solution, a link to the JIRA issue, and to any dependent or related Pull Request.
- [x] Testing instructions have been added in the PR body (for PRs involving changes that are not immediately obvious).
- [x] The developer has manually tested the changes and verified that the changes work

### CODE REVIEW

File: internal/controller/components/kueue/kueue_controller_actions.go [hunk 1]
```diff
@@ -73,6 +77,13 @@ func initialize(_ context.Context, rr *odhtypes.ReconciliationRequest) error {
 		rr.Manifests = append(rr.Manifests, manifestsPath())
 	}
 	rr.Manifests = append(rr.Manifests, kueueConfigManifestsPath())
+
+	// Add template for kueue admin role binding
+	rr.Templates = append(rr.Templates, odhtypes.TemplateInfo{
```
Comment (lburgazzoli) [inline internal/controller/components/kueue/kueue_controller_actions.go:82]: is this actually needed ?

Comment (ugiordan) [reply internal/controller/components/kueue/kueue_controller_actions.go:82]: You're absolutely right! 👍 

The template registration is unnecessary since I'm creating the ClusterRoleBinding programmatically in the `manageKueueAdminRoleBinding` function using Go structs, not template rendering.

I'm opening another PR to removed:
- The template registration code in `initialize()`
- The unused template file `kueue-admin-rolebinding.tmpl.yaml`

Comment (ugiordan) [reply internal/controller/components/kueue/kueue_controller_actions.go:82]: https://github.com/opendatahub-io/opendatahub-operator/pull/2100
